src/main/java/pl/psnc/dei/service/BatchService.java

```diff
@@ -138,7 +139,14 @@ private String getTitle(String recordId) {
 					.filter(obj -> obj.get("dc:title") != null)
 					.findFirst();
 			if (title.isPresent()) {
-				return title.get().getAsObject().get("dc:title").getAsObject().get("@value").getAsString().value();
+				JsonValue dcTitle = title.get().getAsObject().get("dc:title");
+				if (dcTitle.isObject()) {
+					return dcTitle.getAsObject().get("@value").getAsString().value();
+				} else if (dcTitle.isString()) {
+					return dcTitle.getAsString().value();
+				} else {
+					throw new RuntimeException("Cannot read title for record " + recordId + ".");
```

Comment:
Do not throw runtime exception. Probably it would be better to have something special but I think we can just skip the title, log the warning and return recordId just like it is by default when the title is missing.

Reply:
fixed